xapian-letor/tests/api_letor.cc

```diff
@@ -0,0 +1,187 @@
+    delete ranker;
+    ranker = new Xapian::SVMRanker();
+    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker->train_model("", ""));
+    delete ranker;
+
+    return true;
+}
+
+static void
+gen_uniqterms_gt_doclen_db(Xapian::WritableDatabase& db, const string&)
+{
+    Xapian::Document doc;
+    Xapian::TermGenerator termgenerator;
+    termgenerator.set_document(doc);
+    termgenerator.set_stemmer(Xapian::Stem("en"));
+    termgenerator.index_text("Lions, Tigers, Bears and Giraffes", 1, "S");
+    termgenerator.index_text("This paragraph talks about lions and tigers and "
+			     "bears (oh, my!). It mentions giraffes, "
+			     "but that's not really very important. Lions "
+			     "and tigers are big cats, so they must be really "
+			     "cuddly. Bears are famous for being cuddly, at "
+			     "least when they're teddy bears.", 1, "XD");
+    termgenerator.index_text("Lions, Tigers, Bears and Giraffes");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("This paragraph talks about lions and tigers and "
+			     "bears (oh, my!). It mentions giraffes, "
+			     "but that's not really very important. Lions "
+			     "and tigers are big cats, so they must be really "
+			     "cuddly. Bears are famous for being cuddly, at "
+			     "least when they're teddy bears.");
+    db.add_document(doc);
+    doc.clear_terms();
+    termgenerator.index_text("Lions, Tigers and Bears", 1, "S");
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.", 1,
+			     "XD");
+    termgenerator.index_text("Lions, Tigers and Bears");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.");
+    db.add_document(doc);
+}
+
+DEFINE_TESTCASE(ranker1, generated) {
+    Xapian::Ranker *ranker = new Xapian::ListNETRanker();
+    string a = get_database_path("apitest_ranker1",
+				       gen_uniqterms_gt_doclen_db);
+    Xapian::Enquire enquire((Xapian::Database(a)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    string query ="20001 'lions'";
```

Comment:
That's generally good advice, though this is actually a case which `xapian-check-patch` doesn't detect currently (or didn't - I just pushed a fix for it).

But you also shouldn't expect the script to absolve you of having to try to follow the coding conventions as you code.  It's likely it's never going to spot everything, at least with the way it currently works.